### ISSUE TITLE
code intellisense Very slow

### ISSUE THREAD

**Issue by chrmarti** [flowtype/flow-for-vscode#104]
_From @hengkx on April 2, 2017 14:36_

- VSCode Version: Code 1.10.2 (8076a19fdcab7e1fc1707952d652f0bb6c6db331, 2017-03-08T14:02:52.799Z)
- OS Version: Windows_NT ia32 10.0.14393
- Extensions: 

|Extension|Author|Version|
|---|---|---|
|npm-intellisense|christian-kohler|1.3.0|
|path-intellisense|christian-kohler|1.2.0|
|vscode-eslint|dbaeumer|1.2.8|
|vscode-babel-coloring|dzannotti|0.0.4|
|vscode-html-css|ecmel|0.0.33|
|EditorConfig|EditorConfig|0.6.2|
|flow-for-vscode|flowtype|0.5.0|
|auto-close-tag|formulahendry|0.3.10|
|auto-rename-tag|formulahendry|0.0.11|
|vscode-create-saga|hengkx|0.0.2|
|JSDocTagComplete|HookyQR|0.0.2|
|code-beautifier|michelemelluso|1.6.2|
|vscode-scss|mrmlnc|0.6.2|
|vscode-jest|Orta|2.0.5|
|vscode-icons|robertohuertasm|7.4.0|
|sass-indented|robinbentley|1.3.1|
|vscode-react-native|vsmobile|0.3.1|
|JavaScriptSnippets|xabikos|1.4.0|
|ReactSnippets|xabikos|1.2.1|;

---

Steps to Reproduce:

1.
2.

_Copied from original issue: Microsoft/vscode#23818_

**Comment by chrmarti** [comment]
_From @ramya-rao-a on April 2, 2017 19:53_

For which language are you finding intellisense slow?

**Comment by chrmarti** [comment]
_From @hengkx on April 3, 2017 2:56_

@ramya-rao-a  js code intellisense Very slow,use [https://github.com/airtoxin/Electron-React-Boilerplate](url)

**Comment by chrmarti** [comment]
_From @aeschli on April 3, 2017 7:33_

@hengkx Can you attach a code sample that shows the slowness?

**Comment by chrmarti** [comment]
_From @hengkx on April 3, 2017 7:54_

@aeschli custome js code intellisense example `"dg": {
        "prefix": "dg",
        "body": [
            "document.getElementById($1)"
        ],
        "description": "document.getElementById(id)"
    },` small project is normal intellisense ,large project  shows slowness

**Comment by chrmarti** [comment]
_From @aeschli on April 3, 2017 10:59_

Sorry, you have to give more precise steps to show what you are doing and what is slow.
Also screenshots or an animated gif can be helpful.

**Comment by chrmarti** [comment]
_From @hengkx on April 4, 2017 6:56_

@aeschli 
![vs code slow](https://cloud.githubusercontent.com/assets/8358236/24644559/e7970efc-1946-11e7-87a3-4645e60906c6.gif)

**Comment by chrmarti** [comment]
_From @aeschli on April 4, 2017 7:21_

@hengkx Thanks!
Can you run without any extension and see if that makes a difference?
Start code with `code --disable-extensions`

**Comment by chrmarti** [comment]
_From @hengkx on April 4, 2017 13:17_

@aeschli Thanks, This flow-for-vscode	flowtype	0.5.0  disable, code intellisense speed normal

**Comment by fatfatson** [comment]
confirm

**Comment by tgdn** [comment]
Same here, I can barely use flow

**Comment by gskls** [comment]
me too man!

**Comment by tgdn** [comment]
I moved out from Flow and I'm using TypeScript now, much much better. Super fast, automatic imports, code completion, smart variable completion depending on type. It's almost become an IDE!

**Comment by mhnewsuk** [comment]
I am experiencing the same issue. It's really really slow on Windows. I have colleagues sitting next to me on a Mac and on Ubuntu and they're fine. Why is it so slow on Windows?

**Comment by massimonewsuk** [comment]
Has anyone still got this issue with v0.67.0?

**Comment by ferrao** [comment]
I have given up...

**Comment by orta** [comment]
I've been doing some work in jest with flow + vscode using the technique in https://github.com/flowtype/flow-for-vscode/issues/211 and it worked well

**Comment by massimonewsuk** [comment]
@orta are you on windows?

**Comment by orta** [comment]
Nope, ah this is a windows issue - I don't think this issue should be on the vscode extension at all, seems like a problem with flow itself.

**Comment by massimonewsuk** [comment]
You're probably right

**Comment by rssfrncs** [comment]
have any other windows users tried [this language server extension](https://github.com/jbachhardie/vscode-flow-ls)? It seems to be faster than the other flow extensions (still not as fast as on a Mac).

**Comment by thymikee** [comment]
@rssfrncs have you tried setting up `"useLSP": true` in your config? It uses the same language server under the hood.

**Comment by Macstyg** [comment]
The same problem for me on Mac. Rebooting VSCode time from time helps, but this experience is so awful. I'm also working on another project In parallel with React and Typescript and TS works like a charm!

**Comment by samuelhnrq** [comment]
Necrobumping this issue 2 years later, not sure if same issue, already disabled vscode built-in javascript and typescript extensions, tried enabling and disabling LSP, adding more things to ignore, removing [lib] section from the config, nothing seems to work the autocomplete speed is absolutely abysmal (several seconds)

https://user-images.githubusercontent.com/1809521/129070411-1317e38d-869f-480a-ac31-d48b92a224b0.mp4

**Comment by Athelian** [comment]
@DimaLapshyn I cannot even run a flow-server in vscode as the extension crashes every minute or so and then bricks my hard drive re-parsing all the files. 
Do you think there would be benefit of running a flow app in WSL2 even without running the extension other than speeding up `flow check` (something my project needs to run)?

**Comment by Athelian** [comment]
@DimaLapshyn I tried both in WSL2 and on a full Linux dist, the difference is amazing! Intellisense is lightning fast and what was before a 5-10 minute flow check time is now under 30 seconds. Thank you so much.

**Comment by demoh2019** [comment]
+1

**Comment by Py3ik** [comment]
+1

**Comment by Brianzchen** [comment]
Can confirm in 2023 flow on windows stand alone is still very slow and only retriggers rechecks upon saving. Through with WSL works like a charm and just as fast as MacOS
